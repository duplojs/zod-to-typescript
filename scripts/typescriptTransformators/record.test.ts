import { ZodToTypescript } from "@scripts/index";
import { z as zod } from "zod";

describe("record", () => {
	it("<number, string>", () => {
		const result = ZodToTypescript.convert(
			zod.record(zod.number(), zod.string()),
		);

		expect(result).toMatchSnapshot();
	});

	it("<string, string>", () => {
		const result = ZodToTypescript.convert(
			zod.record(zod.string()),
		);

		expect(result).toMatchSnapshot();
	});

	it("<enum, string>", () => {
		const result = ZodToTypescript.convert(
			zod.record(zod.enum(["toto", "tata"]), zod.string()),
		);

		expect(result).toMatchSnapshot();
	});

	it("<string?, string | undefined>", () => {
		const result = ZodToTypescript.convert(
			zod.record(zod.string().optional()),
		);

		expect(result).toMatchSnapshot();
	});
});
