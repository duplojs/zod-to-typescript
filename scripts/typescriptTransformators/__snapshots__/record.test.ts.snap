// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`record > <enum, string> 1`] = `"type Zod2ts_2_duplojs = Record<"toto" | "tata", string>;"`;

exports[`record > <number, string> 1`] = `"type Zod2ts_0_duplojs = Record<number, string>;"`;

exports[`record > <string, string> 1`] = `"type Zod2ts_1_duplojs = Record<string, string>;"`;

exports[`record > <string?, string | undefined> 1`] = `"type Zod2ts_3_duplojs = Partial<Record<string, string | undefined>>;"`;
